static/app/views/starfish/views/webServiceView/endpointOverview/index.tsx

```diff
@@ -179,218 +143,192 @@ export default function EndpointOverview() {
+                              }}
+                            />
+                          </MiniChartPanel>
+                          <MiniChartPanel title={t('Throughput')}>
+                            <Chart
+                              statsPeriod={(statsPeriod as string) ?? '24h'}
+                              height={80}
+                              data={results?.[0] ? [results?.[0]] : []}
+                              start=""
+                              end=""
+                              loading={loading}
+                              utc={false}
+                              stacked
+                              isLineChart
+                              disableXAxis
+                              definedAxisTicks={2}
+                              chartColors={[theme.charts.getColorPalette(0)[0]]}
+                              grid={{
+                                left: '0',
+                                right: '0',
+                                top: '8px',
+                                bottom: '16px',
+                              }}
+                            />
+                          </MiniChartPanel>
+                        </Fragment>
+                      );
+                    }}
+                  </EventsRequest>
+                </ChartsContainerItem2>
+              </ChartsContainer>
+            </StyledRow>
+            <SubHeader>{t('Sample Events')}</SubHeader>
+            <TransactionSamplesTable eventView={eventView} />
+            <SegmentedControlContainer>
+              <SegmentedControl
+                size="xs"
+                aria-label={t('Filter Spans')}
+                value={state.spansFilter}
+                onChange={key => setState({...state, spansFilter: key})}
+              >
+                <SegmentedControl.Item key="">{t('All Spans')}</SegmentedControl.Item>
+                <SegmentedControl.Item key="http">{t('http')}</SegmentedControl.Item>
+                <SegmentedControl.Item key="db">{t('db')}</SegmentedControl.Item>
+              </SegmentedControl>
+            </SegmentedControlContainer>
+            <SpanMetricsTable filter={state.spansFilter} transaction={transaction} />
+            <FacetInsights eventView={eventView} />
+          </Layout.Main>
+        </Layout.Body>
+      </Layout.Page>
+    </PageFiltersContainer>
   );
 }
 
 function SpanMetricsTable({
   filter,
   transaction,
 }: {
-  filter: string;
+  filter: ModuleName;
   transaction: string | undefined;
 }) {
   const location = useLocation();
   const pageFilter = usePageFilters();
 
-  const {
-    isLoading: isTableDataLoading,
-    data: tableData,
-    isRefetching: isTableRefetching,
-  } = useQueryMainTable({transaction: (transaction as string) ?? '', limit: 8});
+  const queryConditions = buildQueryConditions(filter || ModuleName.ALL, location);
+  if (transaction) {
+    queryConditions.push(`transaction = '${transaction}'`);
```

Comment:
I'm not entirely sure if the transaction's HTTP method is available in span metrics yet, but can we just add a TODO here to include that in the query conditions?

Reply:
done!